internal/api/controllers/private/runsCancelActions.go

```diff
@@ -0,0 +1,59 @@
+package private
+
+import (
+	"net/http"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/api/dispatch"
+	"playbook-dispatcher/internal/common/model/generic"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/google/uuid"
+)
+
+func CancelInputV2GenericMap(
+	cancelInput CancelInputV2,
+	account string,
+	runId *uuid.UUID,
+) generic.CancelInput {
+	orgIdString := string(cancelInput.OrgId)
+	principal := string(cancelInput.Principal)
+
+	result := generic.CancelInput{
+		Account:   account,
+		RunId:     runId,
+		OrgId:     &orgIdString,
+		Principal: &principal,
+	}
+
+	return result
+}
+
+func runCancelError(code int) *RunCanceled {
+	return &RunCanceled{
+		Code: code,
+		// TODO report error
+	}
+}
+
+func handleRunCancelError(err error) *RunCanceled {
+	if _, ok := err.(*dispatch.RunNotFoundError); ok {
+		return runCancelError(http.StatusNotFound)
+	}
+
+	if _, ok := err.(*dispatch.RecipientNotFoundError); ok {
+		return runCancelError(http.StatusConflict)
+	}
+
+	if _, ok := err.(*dispatch.RunCancelNotCancelableError); ok {
+		return runCancelError(http.StatusExpectationFailed)
```

Comment:
This HTTP status code has specific meaning related to the use of the `Expect` HTTP header. 409 is a good status for this case also.